python/pyarrow/tests/test_io.py

```diff
@@ -710,6 +710,15 @@ def test_non_cpu_buffer(pickle_module):
     # Sliced buffers with same address
     assert buf_on_gpu_sliced.equals(cuda_buf[2:4])
 
+    # Testing copying buffer
+    mm = ctx.memory_manager
+    for dest in [mm, mm.device]:
+        buf2 = cuda_buf.copy(dest)
+        cuda_buf2 = cuda.CudaBuffer.from_buffer(buf2)
+        assert cuda_buf2.size == cuda_buf.size
+        assert cuda_buf2.copy_to_host()[:] == b'testing'
+        assert cuda_buf2.device == mm.device
```

Comment:
```suggestion
        assert buf2.device == mm.device
```

To test the actual resulting Buffer of the `copy` method?

Reply:
@jorisvandenbossche Ah yes, because regular buffers also have a device property. I used the converted CudaBuffer because I was confused on that front.